lib/dialects/abstract/query-generator.js

```diff
@@ -1754,10 +1754,13 @@ var QueryGenerator = {
    */
   addLimitAndOffset: function(options, model) {
     var fragment = '';
-    if (options.offset && !options.limit) {
+    var limitExist = typeof options.limit !== "undefined" && options.limit !== null;
+    var offsetExit = typeof options.offset !== "undefined" && options.offset !== null;
+
+    if (offsetExit && !limitExist) {
       fragment += ' LIMIT ' + this.escape(options.offset) + ', ' + 10000000000000;
-    } else if (options.limit) {
-      if (options.offset) {
+    } else if (limitExist) {
+      if (offsetExit) {
```

Comment:
You can make this all a bit cleaner by using `options.limit != null`, a oddity of javascript allows you to check both undefined and null with that.